File: packages/python.org/libraries/lodash/libs/math.py
"""
math
"""

import math


def add(a_value, b_value):
    """
    add
    """
    return a_value + b_value


def ceil(number):
    """
    ceil
    """
    return math.ceil(number)


def divide(a_value, b_value):
    """
    divide
    """
    return a_value / b_value


def floor(number):
    """
    floor
    """
    return math.floor(number)


def max(numbers):
    """
    max
    """
    if len(numbers) == 0:
        return
    max_value = float("-inf")
    for number in numbers:
        if number > max_value:
            max_value = number
    return max_value


def max_by(objects, iteratee):
    """
    max_by
    """
    if len(objects) == 0:
        return
    numbers = []
    for obj in objects:
        number = obj[iteratee]
        numbers.append(number)
    return max(numbers)


def mean(numbers):
    """
    mean
    """
    length = len(numbers)
    if length == 0:
        return 0
    sum_value = sum(numbers)
    return sum_value / length


def mean_by(objects, iteratee):
    """
    mean_by
    """
    if len(objects) == 0:
        return
    numbers = []
    for obj in objects:
        number = obj[iteratee]
        numbers.append(number)
    return mean(numbers)


def min(numbers):
    """
    min
    """
    if len(numbers) == 0:
        return
    min_value = float("inf")
    for number in numbers:
        if number < min_value:
            min_value = number
    return min_value


def min_by(objects, iteratee):
    """
    min_by
    """
    if len(objects) == 0:
        return
    numbers = []
    for obj in objects:
        number = obj[iteratee]
        numbers.append(number)
    return min(numbers)


def multiply(a_value, b_value):
    """
    multiply
    """
    return a_value * b_value


def round_number(number, precision=0):
    """
    round_number
    """
    return round(number, precision)


def subtract(a_value, b_value):
    """
    subtract
    """
    return a_value - b_value


# Recursion
def sum(numbers):
    """
    sum
    """
    if len(numbers) == 0:
        return 0
    first = numbers.pop(0)
    return first + sum(numbers)


def sum_by(objects, iteratee):
    """
    sum_by
    """
    if len(objects) == 0:
        return
    numbers = []
    for obj in objects:
        number = obj[iteratee]
        numbers.append(number)
    return sum(numbers)

File: packages/go.dev/libraries/utils/lodash/math/math.go
package math

import "math"

func Add(a float64, b float64) float64 {
	return a + b
}

func Ceil(number float64) float64 {
	return math.Ceil(number)
}

func Divide(a float64, b float64) float64 {
	return a / b
}

func Floor(number float64) float64 {
	return math.Floor(number)
}

func Max(array []float64) float64 {
	var maximum float64 = math.Inf(-8)
	for i := 0; i < len(array); i++ {
		var number float64 = array[i]
		if number > maximum {
			maximum = number
		}
	}
	return maximum
}

func Mean(array []float64) float64 {
	if len(array) == 0 {
		return 0
	}
	var _sum float64 = Sum(array)
	var mean float64 = _sum / float64(len(array))
	return mean
}

func Min(array []float64) float64 {
	var minimum float64 = math.Inf(8)
	for i := 0; i < len(array); i++ {
		var number float64 = array[i]
		if number < minimum {
			minimum = number
		}
	}
	return minimum
}

func Multiply(a float64, b float64) float64 {
	return a * b
}

func Round(number float64) float64 {
	return math.Round(number)
}

func Subtract(a float64, b float64) float64 {
	return a - b
}

func Sum(array []float64) float64 {
	var sum float64 = 0
	for i := 0; i < len(array); i++ {
		var number float64 = array[i]
		sum += number
	}
	return sum
}
